src/main/java/crawlercommons/filters/basic/BasicURLNormalizer.java

```diff
@@ -143,6 +149,15 @@ public String filter(String urlString) {
         try {
             url = new URL(urlString);
         } catch (MalformedURLException e) {
+            if (!hasSchemePattern.matcher(urlString).find()) {
+                // no protocol/scheme : try to prefix http://
+                try {
+                    url = new URL("http://" + urlString);
+                } catch (MalformedURLException e1) {
+                }
+            }
+        }
+        if (url == null) {
             LOG.info("Malformed URL {}", urlString);
```

Comment:
Not related to your code change, but logging at info level here seems wrong - I would change to debug

Reply:
Ok, done.